open_instruct/utils.py

```diff
@@ -0,0 +1,338 @@
+        default="main",
+        metadata={"help": "The specific model version to use (can be a branch name, tag name or commit id)."},
+    )
+    use_flash_attn: bool = field(
+        default=True,
+        metadata={"help": "Whether to use flash attention in the model training"},
+    )
+    # cache_dir: Optional[str] = field(
+    #     default=None,
+    #     metadata={"help": "Where do you want to store the pretrained models downloaded from huggingface.co"},
+    # )
+    use_slow_tokenizer: bool = field(
+        default=True,
+        metadata={"help": "Whether to use one of the slow tokenizer or not (which is then fast tokenizer)."},
+    )
+    model_revision: str = field(
+        default="main",
+        metadata={"help": "The specific model version to use (can be a branch name, tag name or commit id)."},
+    )
+    # token: str = field(
+    #     default=None,
+    #     metadata={
+    #         "help": (
+    #             "The token to use as HTTP bearer authorization for remote files. If not specified, will use the token "
+    #             "generated when running `huggingface-cli login` (stored in `~/.huggingface`)."
+    #         )
+    #     },
+    # )
+    trust_remote_code: bool = field(
+        default=False,
+        metadata={
+            "help": (
+                "Whether or not to allow for custom models defined on the Hub in their own modeling files. This option"
+                "should only be set to `True` for repositories you trust and in which you have read the code, as it will "
+                "execute code present on the Hub on your local machine."
+            )
+        },
+    )
+    # torch_dtype: Optional[str] = field(
+    #     default="auto",
+    #     metadata={
+    #         "help": (
+    #             "Override the default `torch.dtype` and load the model under this dtype. If `auto` is passed, the "
+    #             "dtype will be automatically derived from the model's weights."
+    #         ),
+    #         "choices": ["auto", "bfloat16", "float16", "float32"],
+    #     },
+    # )
+    low_cpu_mem_usage: bool = field(
+        default=False,
+        metadata={
+            "help": (
+                "It is an option to create the model as an empty shell, then only materialize its parameters when the pretrained weights are loaded. "
+                "set True will benefit LLM loading time and RAM consumption."
+            )
+        },
+    )
+    dataset_name: Optional[str] = field(
+        default=None, metadata={"help": "The name of the dataset to use (via the datasets library)."}
+    )
+    dataset_config_name: Optional[str] = field(
+        default=None, metadata={"help": "The configuration name of the dataset to use (via the datasets library)."}
+    )
+    train_file: Optional[str] = field(default=None, metadata={"help": "The input training data file (a json/jsonl file)."})
+    # max_train_samples: Optional[int] = field(
```

Comment:
why remove this?

Reply:
Similar to above, not in `finetune.py`. Can add it in.